docs/configuration.md

```diff
@@ -152,7 +152,7 @@ of the most common options to set are:
   <td><code>spark.driver.memory</code></td>
   <td>1g</td>
   <td>
-    Amount of memory to use for the driver process, i.e. where SparkContext is initialized, in MiB 
+    Amount of memory to use for the driver process, i.e. where SparkContext is initialized, in bytes 
```

Comment:
Check the config code here.
https://github.com/apache/spark/blob/99d2e4e00711cffbfaee8cb3da9b6b3feab8ff18/core/src/main/scala/org/apache/spark/internal/config/package.scala#L40-L43

Reply:
why MiB is wrong btw?

Reply:
MiB is correct in both cases, according to the code. I believe this PR should be closed.

Reply:
@xuanyuanking @HyukjinKwon @srowen thanks for your reply. I've also noticed the code above, but the 'DRIVER_MEMORY' and 'EXECUTOR_MEMORY' in the config/package.scala never used, maybe this is for future usage I think.  The code below shows how the conf is used for now, please take a look.
https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/SparkContext.scala#L465

https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/util/Utils.scala#L1130

https://github.com/ivoson/spark/blob/master/launcher/src/main/java/org/apache/spark/launcher/SparkSubmitCommandBuilder.java#L265

Reply:
If we start a job with conf spark.executor.memory=1024 or spark.driver.memory=1024, it means 1024 bytes for now. So I think we should update the doc to avoid confusion for spark users.

Reply:
I think I got the point you want to report @ivoson, IIUC, this is a bug in the code not in doc, we should also make `spark.driver.memory=1024` with the unit of MiB, maybe this change the original behavior and we can announce in migrate guide? cc @srowen @HyukjinKwon.

Reply:
Oh, hm.

```
$ ./bin/spark-shell --master local --driver-memory=1024
Error occurred during initialization of VM
Too small initial heap
$ ./bin/spark-shell --master local --conf spark.driver.memory=1024
Error occurred during initialization of VM
Too small initial heap
```

Although the option referenced above is used, it's used in YARN and K8S, not in this code path.

Paging @vanzin for an opinion? I'd prefer to change docs rather than behavior but it looks like we might have divergent behavior here.